pkg/tfcli/refresh.go

```diff
@@ -103,52 +104,64 @@ func (c *Client) importResource(ctx context.Context, id string) (model.RefreshRe
 // RefreshResult.State is non-nil and holds the fresh Terraform state iff
 // RefreshResult.Completed is true.
 func (c *Client) observe(ctx context.Context) (model.RefreshResult, error) {
-	result := model.RefreshResult{}
 	// try to save the given state
 	if err := c.storeStateInWorkspace(); err != nil {
-		return result, err
+		return model.RefreshResult{}, err
 	}
 	// now try to run the refresh pipeline synchronously
 	if err := c.syncPipeline(ctx, true, "sh", "-c",
-		fmt.Sprintf("%s apply -refresh-only -auto-approve -input=false && %s plan -detailed-exitcode -refresh=false -input=false",
+		fmt.Sprintf("%s apply -refresh-only -auto-approve -input=false && %s plan -detailed-exitcode -refresh=false -input=false -json",
 			pathTerraform, pathTerraform)); err != nil {
-		return result, err
+		return model.RefreshResult{}, err
 	}
 	code := c.pInfo.GetProcessState().ExitCode()
 	stdout, err := c.pInfo.StdoutAsString()
 	if err != nil {
-		return result, err
+		return model.RefreshResult{}, err
+	}
+	stderr, err := c.pInfo.StderrAsString()
+	if err != nil {
+		return model.RefreshResult{}, err
 	}
+	// Code 2 means that we need to make a change and that's a valid case for
+	// us, so we don't treat it as an error case.
 	if code != 0 && code != 2 {
-		return result, errors.Errorf(fmtErrRefreshExit, code)
+		return model.RefreshResult{}, errors.Errorf(errFmtObserve, stderr)
```

Comment:
Should we still return exit code as part of the error?

Reply:
done!